### PR TITLE
open and write in byte mode using fileserver

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
This switches to byte mode explicitely in the last few places I could find that
are using the fileserver payload to store files.

We should only ever be reading in the file, and then passing it to the minion,
the unicode locale on the master should not matter.  Then we want to make sure
to write the information in bytes to the filesystem.

### CODE REVIEW

File: salt/fileserver/hgfs.py
Comment (s0undt3ch) [inline salt/fileserver/hgfs.py]: Since you now opened in binary mode, you can only write bytes to the file. Since this is ASCII, it's safe to do `b'ascii string'`. On py3 of course.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/fileserver/hgfs.py
Comment (gtmanfred) [inline salt/fileserver/hgfs.py]: Sorry, i missed reverting this one.

The only ones that needed to be changed was the env_cache which used the Serial output.

Thanks,
Daniel

**Comment by rallytime** [comment]
I have switched the base branch on this pull request from `nitrogen` to `2017.7` in preparation for moving from a code-named release to a version number.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Review by terminalmage — DISMISSED** [review]

File: salt/fileserver/hgfs.py
Comment (terminalmage) [inline salt/fileserver/hgfs.py]: `salt.utils.to_bytes()` is a shorthand for this.

Comment (terminalmage) [inline salt/fileserver/hgfs.py]: Oh you're going to revert it

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@s0undt3ch and @terminalmage Re-review requested.

**Review by terminalmage — APPROVED** [review]

**Comment by cachedout** [comment]
Just waiting on @s0undt3ch for one more review and then we should be set.

**Review by s0undt3ch — APPROVED** [review]

**Comment by rallytime** [comment]
Go Go Jenkins!

**Pull request merged**
